chainstate/src/detail/tests/test_framework/framework.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Sinitsyn, S. Tkach
+
+use chainstate_storage::BlockchainStorageRead;
+use common::{
+    chain::{
+        config::{Builder as ChainConfigBuilder, ChainType},
+        Block, Destination, GenBlock, Genesis, NetUpgrades, OutputPurpose, TxOutput,
+    },
+    primitives::{id::WithId, Amount, Id, Idable},
+};
+use crypto::random::Rng;
+
+use crate::{
+    detail::{
+        tests::{
+            test_framework::{BlockBuilder, TestFrameworkBuilder, TransactionBuilder},
+            TestBlockInfo,
+        },
+        BlockIndex, GenBlockIndex, TimeGetter,
+    },
+    BlockError, BlockHeight, BlockSource, Chainstate, ChainstateConfig,
+};
+
+/// The `Chainstate` wrapper that simplifies operations and checks in the tests.
+pub struct TestFramework {
+    pub chainstate: Chainstate,
+    // TODO: FIXME: Remove?..
```

Comment:
👀

Reply:
I have made several attempts to remove this field, but it is heavily used in the reorg tests and sometimes it will require to almost completely rewrite them. At the moment I feel like doing that isn’t worth it.